Week3/Homework/Js-exercises/cardValidator.js

```diff
@@ -0,0 +1,45 @@
+"use strict";
+
+const cardValidator = function(cardNumber) {
+  //first converts our number into an array of strings then using for of, converts the string array to numbers array
+  if (Number.isInteger(Number(cardNumber)) !== true) {
+    return console.log("invalid characters!");
+  }
+  let strArr = cardNumber.toString().split("");
+  const numArr = [];
+  for (let ele of strArr) {
+    numArr.push(Number(ele));
+  }
+  console.log(numArr.length);
```

Comment:
In general for this question, I would suggest that you do all operations which you require i.e creating arrays, calculating sums etc and then once it is done should you have a consolidated if conditions. Otherwise, it is very hard to read the function 🙂